build/localnet/cluster-manager/main.go

```diff
@@ -31,13 +35,45 @@ var (
 	// autoStakeSkipStakeForValidatorIds is a list of validator ids that should not be auto-staked
 	// it is used to avoid auto-staking the validators that are already staked as part of genesis.
 	autoStakeSkipStakeForValidatorIds = []string{"001", "002", "003", "004"}
+
+	clusterManagerCmd = &cobra.Command{
+		Use:   "cluster-manager",
+		Short: "Start the Pocket Network Cluster Manager service",
+		Long: `Start the Pocket Network Cluster Manager service which listens for and reacts to events coming over the k8s.io API's watch.Interface#ResultChan().
+
+See the following k8s.io documentation for more information:
+- https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1@v0.26.1#ServiceInterface
+- https://pkg.go.dev/k8s.io/apimachinery/pkg/watch#Interface,
+- https://pkg.go.dev/k8s.io/apimachinery/pkg/watch#Event`,
+		Run:  runClusterManagerCmd,
+		Args: cobra.ExactArgs(0),
+	}
 )
 
 func init() {
-	rpcURL = fmt.Sprintf("http://%s:%s", runtime.GetEnv("RPC_HOST", defaults.Validator1EndpointK8S), defaults.DefaultRPCPort)
+	// setup the `remote_cli_url` flag to be consistent with the CLI except for
+	// the default, which is set to the URL of the first k8s validator's RPC
+	// endpoint.
+	clusterManagerCmd.PersistentFlags().StringVar(
+		&flags.RemoteCLIURL,
+		"remote_cli_url",
+		defaults.Validator1EndpointK8SHostname,
+		"takes a remote endpoint in the form of <protocol>://<host>:<port> (uses RPC Port)",
+	)
+
+	// ensure that the env var can override the flag
```

Comment:
I know we've had a lot of issues with viper and flag binding in the past.

I don't have much feedback on this as I'd have to dive in really deep, but have to be honest that I have lost context on where/how this variable binding is working.

Anything I should focus on or provide feedback in particular? If not, LGTM

Reply:
What I think I've learned regarding bindings: cobra takes the default values and handles updating bound flag variables/pointers based on any actual flag values which are present during invocation (i.e. `--remote_cli_url someValue`). Separately from that, viper can track a key of the same name, the value of which (independent of the flag vars/ptrs) will default to its respective config value (if present/applicable) but can be overridden by a respective env var and/or the flag value.

TLDR; the viper/cobra API dynamic is such that:
1. Viper key/values are separate (in memory) from viper flags.
2. Viper key/values can be overridden by cobra flags but not the other way around.
3. For consistency we should either
  - Ensure we're updating flag vars (i.e. bound to a cobra flag) with their respective viper overrides
  - Stop using flag vars (e.g. do `#PersistentFlags.String(...)` instead of `#PersistentFlags.StringVar(&flagVar, ...)` and treat viper as the single source of truth (e.g. `viper.GetString(...)`)